pkg/merger/merger_test.go

```diff
@@ -408,6 +408,71 @@ func Test_MergeAndUpdate(t *testing.T) {
 		})
 	}
 }
+func Test_RecordLastModified(t *testing.T) {
+	cases := []struct {
+		name   string
+		attrs  *storage.ReaderObjectAttrs
+		mets   *Metrics
+		source string
+	}{
+		{
+			name:   "nil attrs; succeeds",
+			attrs:  nil,
+			mets:   &Metrics{},
+			source: "fake-source",
+		},
+		{
+			name:   "nil mets; succeeds",
+			attrs:  &storage.ReaderObjectAttrs{},
+			mets:   nil,
+			source: "fake-source",
+		},
+		{
+			name:   "nil; succeeds",
+			attrs:  nil,
+			mets:   nil,
+			source: "",
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			recordLastModified(tc.attrs, tc.mets, tc.source)
+		})
+	}
+}
+
+func Test_RecordLastModified_RecordMetrics(t *testing.T) {
```

Comment:
I would combine this test and the previous test into one table-driven test. The table might be something like

```
name string
attrs *storage.ReaderObjectAttrs
mets *Metrics
expectRecord bool
expectValue int
```

... then you could check all these cases in one test.

Reply:
https://github.com/GoogleCloudPlatform/testgrid/blob/master/pkg/tabulator/filter_test.go has some examples of these "expect" booleans and expected outputs.